plugins/CoreHome/vue/src/SiteSelector/SiteSelector.adapter.ts

```diff
@@ -82,11 +75,18 @@ export default createAngularJsAdapter<[ITimeoutService]>({
       }
     });
 
+    if (attrs.siteid && attrs.sitename) {
+      vm.modelValue = { id: attrs.siteid, name: Matomo.helper.htmlDecode(attrs.sitename) };
+    } else if (Matomo.idSite) {
+      vm.modelValue = {
+        id: Matomo.idSite,
+        name: Matomo.helper.htmlDecode(Matomo.siteName),
+      };
+    }
```

Comment:
Did you do those changes on purpose? It seems now pre selection in the site selector don't work anymore. You can see that in the siteselector within the usersettings.

Reply:
They were to fix another issue. Seems like the new problem is due to two reasons:

1. The original `vm.modelValue =` in postCreate() wasn't being applied. Moved it to a default() function in the adapter definition so the vue app data would be initiated to that value.
2. It turns out if the `default` value for a Vue property is a factory function, it is applied no matter what, even if an original value is supplied. Now we check if there's a value set first.

Hopefully the fix won't break something elsewhere.

Reply:
UI tests need to be updated (I'll do that if the PR is reviewed, otherwise tomorrow), but it looks like the change worked.